Check for signed-off-by doesn't ignore comments

## Expected Behavior

When using the rule `"signed-off-by": [2, "always", "Signed-off-by:"]`, it doesn't seem to use the parsed commit message when reading in the lines. I would expect the values of `lines` to be `['subject', 'body', 'Signed-off-by: User email@address.com']`.

## Current Behavior

The current behaviour includes the comments that are part of `.git/COMMIT_EDIT_MSG` so it is checking for a 'Signed-off-by' line on the last comment rather than the last actual line of the message, causing it to always fail the check.

## Affected packages

- [x] cli
- [ ] core
- [ ] prompt
- [ ] config-angular

## Possible Solution

Use the correctly parsed input?

## Steps to Reproduce (for bugs)

Make a commit using `git commit -s` with the setting `"signed-off-by": [1, "always", "Signed-off-by:"]`.


## Context

I'm trying to enforce signed commits in our repo.

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

|             Executable | Version |
| ---------------------: | :------ |
| `commitlint --version` | 8.3.5 |
|        `git --version` | 2.21.1 |
|       `node --version` | v10.16.2 |

The commit I'm making looks like this (not done through -m):

```
fix: remove test file

Contributes to: #11

Signed-off-by: Jordan <my@email.com>
```

Which then gives the correct output in the console as:

```
⧗   input: fix: remove test file

Contributes to: #11

Signed-off-by: Jordan <my@email.com>
```

However, it gives the error: `✖   message must be signed off [signed-off-by]`

When looking deeper into the code, I logged out what the value of `lines` is from this file: 

https://github.com/conventional-changelog/commitlint/blob/0e0c68dba40c5b36e0d9af8e685b37664dd3e133/%40commitlint/rules/src/signed-off-by.ts#L10

And it gives the value as:

```
['fix: remove test file','Contributes to: #11','Signed-off-by: Jordan <my@email.com>','# Please enter the commit message for your changes. Lines starting','# with '#' will be ignored, and an empty message aborts the commit.,'#','# On branch actions','# Your branch is up to date with 'origin/actions'.','#','# Changes to be committed:','#       deleted:    test.js','#']
```

So, the values of `last` is `#` instead of the last actual line in the commit. So, it looks like it usually parses the comments out as when I disable this line: https://github.com/conventional-changelog/commitlint/blob/0e0c68dba40c5b36e0d9af8e685b37664dd3e133/%40commitlint/cli/src/cli.js#L175
The output in the console then shows all the comments as well. But, it doesn't use this parsed result when checking the last line for a `Signed-off-by`string.

--- a/@commitlint/rules/src/signed-off-by.test.ts
+++ b/@commitlint/rules/src/signed-off-by.test.ts
@@ -7,6 +7,15 @@ const messages = {
 	without: `test: subject\nbody\nfooter\n\n`,
 	inSubject: `test: subject Signed-off-by:\nbody\nfooter\n\n`,
 	inBody: `test: subject\nbody Signed-off-by:\nfooter\n\n`,
+	withSignoffAndComments: `test: subject
+
+message body
+
+Signed-off-by:
+
+# Please enter the commit message for your changes. Lines starting
+# with '#' will be ignored, and an empty message aborts the commit.
+`,
 };
 
 const parsed = {
@@ -15,6 +24,7 @@ const parsed = {
 	without: parse(messages.without),
 	inSubject: parse(messages.inSubject),
 	inBody: parse(messages.inBody),
+	withSignoffAndComments: parse(messages.withSignoffAndComments),
 };
 
 test('empty against "always signed-off-by" should fail', async () => {
@@ -53,6 +63,16 @@ test('without against "always signed-off-by" should fail', async () => {
 
 test('without against "never signed-off-by" should succeed', async () => {
 	const [actual] = signedOffBy(await parsed.without, 'never', 'Signed-off-by:');
+	const expected = true;
+	expect(actual).toEqual(expected);
+});
+
+test('trailing comments should be ignored', async () => {
+	const [actual] = signedOffBy(
+		await parsed.withSignoffAndComments,
+		'always',
+		'Signed-off-by:'
+	);
 	const expected = true;
 	expect(actual).toEqual(expected);
 });

--- a/@commitlint/rules/src/signed-off-by.ts
+++ b/@commitlint/rules/src/signed-off-by.ts
@@ -7,7 +7,14 @@ export const signedOffBy: SyncRule<string> = (
 	when = 'always',
 	value = ''
 ) => {
-	const lines = toLines(parsed.raw).filter(Boolean);
+	const lines = toLines(parsed.raw).filter(
+		(ln) =>
+			// skip comments
+			!ln.startsWith('#') &&
+			// ignore empty lines
+			Boolean(ln)
+	);
+
 	const last = lines[lines.length - 1];
 
 	const negated = when === 'never';

